guides/source/getting_started.md

```diff
@@ -334,216 +340,650 @@ If we go back to our browser and make a request to
-application. The format for this template can only be `html` and the default
-handler for HTML is `erb`. Rails uses other handlers for other formats.
-`builder` handler is used to build XML templates and `coffee` handler uses
-CoffeeScript to build JavaScript templates. Since you want to create a new
-HTML form, you will be using the `ERB` language which is designed to embed Ruby
-in HTML.
+```erb
+<h1>Articles</h1>
 
-Therefore the file should be called `articles/new.html.erb` and needs to be
-located inside the `app/views` directory of the application.
+<ul>
+  <% @articles.each do |article| %>
+    <li><%= article.title %></li>
+  <% end %>
+</ul>
+```
 
-Go ahead now and create a new file at `app/views/articles/new.html.erb` and
-write this content in it:
+This code will render an `li` element for each of the articles, and that
+element contains the title of the article. But we can't click on the title to
+go to an article yet! To make that happen, we need to use an `a` tag:
 
-```html
-<h1>New Article</h1>
+```erb
+<h1>Articles</h1>
+
+<ul>
+  <% @articles.each do |article| %>
+    <li>
+      <a href='/articles/<%= article.id %>'>
+        <%= article.title %>
+      </a>
+    </li>
+  <% end %>
+</ul>
 ```
 
-When you refresh <http://localhost:3000/articles/new> you'll now see that the
-page has a title. The route, controller, action, and view are now working
-harmoniously! It's time to create the form for a new article.
+This `a` tag will provide us with a link to the specific article. If we go back
+to <http://localhost:3000/>, we'll see that we can now click on the articles:
+
+![Articles list with links](images/getting_started/articles_list_with_links.png)
+
+Clicking either of these links will take us to the relevant article:
+
+![Single Article View](images/getting_started/single_article_view.png)
+
+Now we have been able to link together the `index` and `show` pages in our
+application using a simple `a` tag. What could be simpler than that?
+
+Well, Rails has a method called `link_to` that can make that linking a little
+simpler. Let's use this in `app/views/articles/index.html.erb`:
+
+```erb
+<h1>Articles</h1>
+
+<ul>
+  <% @articles.each do |article| %>
+    <li>
+      <%= link_to article.title, "/articles/#{article.id}" %>
+    </li>
+  <% end %>
+</ul>
+```
+
+There we go, that is now a little bit cleaner. Rails has given us a way to
+shorten this code a little. But what you don't know yet is that this line can
+be made even simpler.
+
+Rails has a feature called _routing helpers_. These are methods that can be
+used to generate route paths like `"/articles/#{article.id}"` programatically.
+We'll use one of these to generate the route for our article. To set this up,
+let's go back to `config/routes.rb` and change this line:
+
+```ruby
+get "/articles/:id", to: "articles#show"
+```
+
+To this:
+
+```ruby
+get "/articles/:id", to: "articles#show", as: :article
+```
```

Comment:
I feel like maybe this is kind of confusing since most of the time you don't type out every route since they're restful. We can say `resources :articles` and get all the routes at once.

I also don't love that we're showing them the wrong way (ie `<%= link_to article.title, "/articles/#{article.id}" %>`) _especially_ since that way _could_ result in a security issue if used incorrectly with another parameter.

Reply:
I agree that it would be confusing -- to those who have already seen the `resources` syntax elsewhere.

What would you reckon about demonstrating that `resources :articles` _does_ exist, and we _could_ use it, but clearly explaining that we're avoiding using it to demonstrate what that line gives us?

Reply:
Reading that again, I am not sure I am making 100% sense with that last sentence. Here's a commit that shows the changes I'm suggesting: https://github.com/rails/rails/pull/38288/commits/7121f157885e3e09f96510b9ec6524a1bc8c555f